connection_pool.go

```diff
@@ -359,10 +361,20 @@ func (cp *connectionPool) Close() {
 		backend.Close()
 	}
 	cp.backends = nil
+	if !cp.closed {
+		close(cp.closedChan)
```

Comment:
Did we ever decide on whether or not it's safe to close a channel while other people are reading against it? It's certainly safe to read from a closed channel, but we received reports of races detected when closing a channel we're reading from immediately.